antenna/crashmover.py

```diff
@@ -137,102 +87,47 @@ def check_health(self, state):
         if hasattr(self.crashpublish, "check_health"):
             self.crashpublish.check_health(state)
 
-    def hb_report_health_stats(self):
-        """Heartbeat function to report health stats."""
-        # The number of crash reports sitting in the work queue; this is a
-        # direct measure of the health of this process--a number that's going
-        # up means impending doom
-        MYMETRICS.gauge("work_queue_size", value=len(self.crashmover_queue))
-
-    def has_work_to_do(self):
-        """Return whether this still has work to do."""
-        work_to_do = len(self.crashmover_pool) + len(self.crashmover_queue)
-        LOGGER.info("work left to do: %s" % work_to_do)
-        # Indicates whether or not we're sitting on crashes to save--this helps
-        # keep Antenna alive until we're done saving crashes
-        return bool(work_to_do)
-
-    def hb_run_crashmover(self):
-        """Spawn a crashmover if there's work to do."""
-        # Spawn a new crashmover if there's stuff in the queue and we haven't
-        # hit the limit of how many we can run
-        if self.crashmover_queue and self.crashmover_pool.free_count() > 0:
-            self.crashmover_pool.spawn(self.crashmover_process_queue)
-
-    def add_crashreport(self, raw_crash, dumps, crash_id):
-        """Add a new crash report to the crashmover queue.
+    @MYMETRICS.timer("crash_handling.time")
+    def handle_crashreport(self, raw_crash, dumps, crash_id):
```

Comment:
this timing was moved here from `crashmover_finish`, because synchronous handling means we don't need to manually calculate it anymore